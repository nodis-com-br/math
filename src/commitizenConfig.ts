const COMMIT_TYPES = {
  feat: {
    description: 'A new feature',
    emoji: '🎸',
    value: 'feat',
  },
  fix: {
    description: 'A bug fix',
    emoji: '🐛',
    value: 'fix',
  },
  refactor: {
    description: 'A code change that neither fixes a bug or adds a feature',
    emoji: '♻️',
    value: 'refactor',
  },
  test: {
    description: 'Adding application tests',
    emoji: '💊',
    value: 'test',
  },
  format: {
    description: 'Code formatting',
    emoji: '💅',
    value: 'format',
  },
  conflicts: {
    description: 'Resolving conflicts',
    emoji: '⚠️',
    value: 'conflicts',
  },
  remove: {
    description: 'Removing unnecessary files',
    emoji: '❌',
    value: 'remove',
  },
  docs: {
    description: 'Documentation only changes',
    emoji: '📚',
    value: 'docs',
  },
  config: {
    description: "Changing project's configuration",
    emoji: '🔧',
    value: 'config',
  },
  security: {
    description: 'A security fix',
    emoji: '🔒',
    value: 'security',
  },
  ci: {
    description: 'CI related changes',
    emoji: '🎡',
    value: 'ci',
  },
};

export const commitizenConfig: {
  list: string[];
  maxMessageLength: number;
  minMessageLength: number;
  questions: string[];
  scopes: string[];
  types: any;
} = {
  list: Object.keys(COMMIT_TYPES),
  maxMessageLength: 100,
  minMessageLength: 0,
  questions: [
    'type',
    'scope',
    'subject',
    'body',
    'breaking',
    'issues',
    'lerna',
  ],
  scopes: [],
  types: COMMIT_TYPES,
};
