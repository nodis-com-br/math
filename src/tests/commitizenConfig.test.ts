import { commitizenConfig } from '../commitizenConfig';

describe('commitizenConfig', () => {
  test('should match snapshot', () => {
    expect(commitizenConfig).toMatchSnapshot();
  });
});
