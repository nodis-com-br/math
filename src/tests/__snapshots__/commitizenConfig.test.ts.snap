// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commitizenConfig should match snapshot 1`] = `
Object {
  "list": Array [
    "feat",
    "fix",
    "refactor",
    "remove",
    "test",
    "packages",
    "config",
    "docs",
    "ci",
    "analytics",
  ],
  "maxMessageLength": 100,
  "minMessageLength": 0,
  "questions": Array [
    "type",
    "scope",
    "subject",
    "body",
    "breaking",
    "issues",
    "lerna",
  ],
  "scopes": Array [],
  "types": Object {
    "analytics": Object {
      "description": "Adding analytics or tracking code",
      "emoji": "📈",
      "value": "analytics",
    },
    "ci": Object {
      "description": "CI related changes",
      "emoji": "👷",
      "value": "ci",
    },
    "config": Object {
      "description": "Changing configuration files",
      "emoji": "🔧",
      "value": "config",
    },
    "docs": Object {
      "description": "Writing docs",
      "emoji": "📝",
      "value": "docs",
    },
    "feat": Object {
      "description": "Introducing new features",
      "emoji": "✨",
      "value": "feat",
    },
    "fix": Object {
      "description": "Fixing a bug",
      "emoji": "🐛",
      "value": "fix",
    },
    "packages": Object {
      "description": "Managing dependencies",
      "emoji": "📦",
      "value": "packages",
    },
    "refactor": Object {
      "description": "Refactoring code",
      "emoji": "♻️",
      "value": "refactor",
    },
    "remove": Object {
      "description": "Removing code or files",
      "emoji": "🔥",
      "value": "remove",
    },
    "test": Object {
      "description": "Creating or updating tests",
      "emoji": "✅",
      "value": "test",
    },
  },
}
`;
